src/array_api_stubs/_draft/data_type_functions.py

```diff
@@ -13,10 +13,15 @@
 
 
 def astype(
-    x: array, dtype: dtype, /, *, copy: bool = True, device: Optional[device] = None
+    x: array,
+    dtype_or_kind: Union[dtype, str],
```

Comment:
I'd prefer to keep the name as `dtype`, the longer one is quite unwieldy. Also if anyone implemented this without using positional-only, then they may not be able to change the name.